### ISSUE TITLE
omi.keytab doesn't work in crontab on Ubuntu 14/16 if we add krb5.keytab after installing omi instead of before installing omi

### ISSUE THREAD

**Issue by JumpingYang001** [Microsoft/omi#405]
Following scenario works on Redhat 6/7 and CentOS 6/7, but doesn't work on Ubuntu 14/16.
1.Install OMI without /etc/krb5.keytab
2.Add /etc/krb5.keytab
3.Wait about 1 minute, crontab generate /etc/opt/omi/creds/omi.keytab
4.Query omiserver works fine without restarting omiserver.

**Environment: non64-ubun16-04 and non64-ubun14-03**
1.Install OMI without /etc/krb5.keytab
2.Add /etc/krb5.keytab
3.Wait about 1 minute, crontab **doesn't** generate /etc/opt/omi/creds/omi.keytab
4.Query omiserver Kerberos **doesn't work** no matter restarting omiserver or not.

**OMI Build:** 1.3.0-92 and 1.4.0 omi in https://github.com/Microsoft/SCXcore-kits/tree/omi-v1.4.0-0/release

I try add `echo `[ /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab ] && echo pass`  >> /root/myscript.log 2>&1` in `crontab -e`, and from myscript.log, it shows the value of `[ /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab ] ` is always `false` while it is `true` on Putty Console, thanks.

I echo the `env` in crontab, it shows `HOME=/root LOGNAME=root PATH=/usr/bin:/bin LANG=en_US.UTF-8 SHELL=/bin/sh PWD=/root`

I echo the `env` in Putty Console, it shows
```
XDG_SESSION_ID=10250
TERM=xterm
SHELL=/bin/bash
SSH_CLIENT=172.18.28.127 53802 22
SSH_TTY=/dev/pts/3
USER=root
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
MAIL=/var/mail/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
PWD=/root
LANG=en_US.UTF-8
SHLVL=1
HOME=/root
LOGNAME=root
SSH_CONNECTION=172.18.28.127 53802 10.226.222.82 22
LESSOPEN=| /usr/bin/lesspipe %s
XDG_RUNTIME_DIR=/run/user/0
LESSCLOSE=/usr/bin/lesspipe %s %s
_=/usr/bin/env
```

**Comment by yakman2020** [comment]
This does not look like a supported scenario.

**Comment by yakman2020** [comment]
How do you add the /etc/krb5.keytab?

**Comment by lilyfang** [comment]
Jumping, why do you need manually add /etc/krb5.keytab?

**Comment by JumpingYang001** [comment]
@yakman2020 @lilyfang , ho sorry, I didn't clear how I added the file, in fact, we added keytab file using `realm --verbose join contoso.com -U Administrator`, is it a support scenario? thanks

**Comment by yakman2020** [comment]
That should be fine. 
Did you try touching the /etc/krb5.keytab to update the date?
Did you try running the crontab line via sh?

**Comment by yakman2020** [comment]
Does the etc/opt/omi/creds keytab file exist? If not, you will have to manually to execute the script line without the condition.

**Comment by yakman2020** [comment]
Looks like a real scenario.  I will generate a fix and new build. The solution is the cronjob should succeed and run the script if the omi.keytab doesn't exist.  It shouldn't take long to get that to you. 

Good catch.

**Comment by JumpingYang001** [comment]
@yakman2020 , in other word, the crontab on Ubuntu won't never hit ktstrip even the crontab will run every minute, thanks for fixing it!

**Comment by JumpingYang001** [comment]
no matter the /etc/krb5.keytab exist or not, it does never hit ktstrip in crontab, thanks.

**Comment by lilyfang** [comment]
Thanks for the clarification and nice catch, Jumping!

**Comment by JumpingYang001** [comment]
Fixed in #408 .
